### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,3 +29,142 @@ root
...
+
+To update the Rollup cell, we build a tx to consume the old one and generate a new Rollup cell as the tx's output.
+We also need to provide a args in the tx's witness: RollupAction, which is a structure that contains layer-2 block and Merkle proof.
+The state validator will make sure the state transition of the Rollup cell is valid by verifying these proofs.
+
+The behaviors of Rollup are defined as enumerated type `RollupAction`:
+- `RollupSubmitBlock`, submit a layer-2 block
+  - The layer-2 transactions, deposits, and withdrawals are included in a layer-2 block structure. We won't verify txs' and withdrawals' signatures on-chain since we are using the optimistic mechanism.
+  - Deposits cells are collected as inputs, and the action converts these deposit cells to custodian cells to complete the deposit.
+- `RollupEnterChallenge`, A challenger submit a challenging target(transaction, deposit, withdrawal) to halt the rollup.
```
Comment (zeroqn) [inline README.md:65]: Deposit?

File: README.md [hunk 2]
```diff
@@ -29,3 +29,142 @@ root
...
+Stake lock can be unlocked in two paths:
+
+1. Unlock by the submitter after `args.stake_block_number`'s block is finalized.
+2. Unlock by the challenger in the `RollupRevert` action.
+
+### Deposit lock
+
+A layer1 user can join the Rollup by creating a deposit cell. The Godwoken collects deposit cells from the layer1 blockchain and put them into the inputs of the tx that submit layer-2 block.
+
+The sender can unlock a deposit cell if the deposit is not processed by Godwoken.
```
Comment (zeroqn) [inline README.md:87]: After `cancel_timeout`.

File: README.md [hunk 3]
```diff
@@ -29,3 +29,142 @@ root
...
+
+### Challenge lock
+
+When a Godwoken node found that an invalid state exists in the Rollup, the node can send the `RollupEnterChallenge` action to the Rollup cell and generate a challenging cell.
+
+A challenge cell must set a challenging target in its lock args `ChallengeLockArgs`. The challenging target can be a layer-2 transaction or a withdrawal request.
+
+If the challenging cell hasn't been canceled during a maturity time, the challenger can execute the `RollupRevert` action on the Rollup cell and take stake cells which send by reverted block submitters as rewards.
+
+If the challenge target is valid. Other nodes can cancel this challenge by executing the `RollupCancelChallenge` action, the challenging cell must be included in the tx.inputs.
```
Comment (zeroqn) [inline README.md:116]:
```suggestion
If the challenge target is invalid. Other nodes can cancel this challenge by executing the `RollupCancelChallenge` action, the challenging cell must be included in the tx.inputs.
```

**Review by zeroqn — COMMENTED** [review]
Some suggestions

**Review by Flouse — APPROVED** [review]
LGTM

**Pull request merged**